File: gear-ecommerce-front/src/material/model/VnGeographyProvince.ts
import {AbstractModel} from "./main/AbstractModel";
import {SubCatalog} from "./SubCatalog";
import {VnGeographyDistrict} from "./VnGeographyDistrict";

export interface VnGeographyProvince extends AbstractModel {
    name: string;
    code: string;
    district?: VnGeographyDistrict[];
}

File: gear-ecommerce-back/src/main/java/tht/closure/operator/model/entity/VnGeographyProvince.java
package tht.closure.operator.model.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "t_vn_geography_province")
@Getter
@Setter
public class VnGeographyProvince extends AbstractEntity{

    private String name;

    private String code;

    @OneToMany(mappedBy = "province")
    private List<VnGeographyDistrict> districts = new ArrayList<>();

    @OneToMany(mappedBy = "province")
    private List<VnGeographyWard> wards = new ArrayList<>();

}
